tools/java_api/src/test/java/com/kuzudb/DatabaseTest.java

```diff
@@ -15,35 +15,35 @@ public class DatabaseTest extends TestBase {
     void DBCreationAndDestroyWithArgs() {
         try {
             String dbPath = tempDir.toFile().getAbsolutePath();
-            Database database = new Database(
+            try (Database database = new Database(
```

Comment:
I think you should add seperate `catch` for database creation. If your database creation failed, your test case should be aware that it is caused by database creation failure. 
So something like:
```java
String dbPath;
try {
  String dbPath = tempDir.toFile().getAbsolutePath();
} catch(Exception e) {
  fail("Cannot get database path: " + e.getMessage();
}

try (Database database = new Database(...){
}
catch(e){
  fail("DBCreationAndDestroyWithArgs failed: " + e.getMessage();
}
```

Reply:
Note that this applies to all functions in the same file